src/helpers/privacy_budget.rs

```diff
@@ -0,0 +1,80 @@
+/// This module implements privacy budget using a redis cache
+/// It expects a vector of encrypted blobs alongwith budget to be consumed.
+/// After increasing the budget consumed by all the blobs, it returns blobs which are below ``TOTAL_PRIVACY_BUDGET``
+
+use redis::Commands;
+use std::env;
+
+const TOTAL_PRIVACY_BUDGET: i32 = 100;
+
+#[must_use]
+pub fn get_eligible_packets_with_budget(packets: &[String], budget: i32) -> Vec<String> {
```

Comment:
Taking packets as a slice of strings is likely to be quite inconvenient.  What this probably needs is a new trait that produces a sequence of bytes that is used as a key to the database.

Ideally, this takes an iterator and produces an iterator so that this acts as filtering function on that iterator.  Alternatively, if you wanted to be really smart, this could produce a closure that you would pass to `Iterator::filter`.

Reply:
Since we need to keep a track of both eligible(to return) and ineligible packets (to later decrement their budget), chose to keep it as a normal iterator. Also have changed the input type of Ciphertext since we would be expecting match keys to be sent to Privacy Budget module